src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -591,11 +603,11 @@ object AppDefinition extends GeneralPurposeCombinators {
   def residentUpdateIsValid(from: AppDefinition): Validator[AppDefinition] = {
     val changeNoVolumes =
       isTrue[AppDefinition]("Persistent volumes can not be changed!") { to =>
-        val fromVolumes = from.persistentVolumes
-        val toVolumes = to.persistentVolumes
+        val fromVolumes = from.persistentVolumesAndMounts
+        val toVolumes = to.persistentVolumesAndMounts
         def sameSize = fromVolumes.size == toVolumes.size
-        def noChange = from.persistentVolumes.forall { fromVolume =>
-          toVolumes.find(_.containerPath == fromVolume.containerPath).contains(fromVolume)
+        def noChange = fromVolumes.forall { fromVolume =>
+          toVolumes.find(_.mount.mountPath == fromVolume.mount.mountPath).contains(fromVolume)
```

Comment:
Can you speak some words to this, too? What was the motivation for this change?

Reply:
It is going to be true in case of pods. In case of apps, `mountPath` (known as `containerPath` previously) is used when creating volumes. Please refer to `OfferOperationFactory.createVolumes`.